app/Services/PirepService.php

```diff
@@ -719,4 +726,101 @@ public function setPilotState(User $pilot, Pirep $pirep)
 
         event(new UserStatsChanged($pilot, 'airport', $previous_airport));
     }
+
+    public function handleDiversion(Pirep $pirep): void
+    {
+        // Return if diversion handling is disabled
+        if (!setting('pireps.handle_diversion', false)) {
+            return;
+        }
+
+        $diversion_airport_id = $pirep->fields->where('slug', 'diversion-airport')->first()?->value;
+
+        // Return if no diversion
+        if (!$diversion_airport_id) {
+            return;
+        }
+
+        $diversion_airport = $this->airportRepo->findWithoutFail($diversion_airport_id);
+
+        // Return if diversion airport not found and airport lookup is disabled
+        if (!$diversion_airport && !setting('general.auto_airport_lookup', false)) {
+            return;
+        }
+
+        if (!$diversion_airport) {
+            $diversion_airport = $this->airportSvc->lookupAirportIfNotFound($diversion_airport_id);
+        }
+
+        // Return if we still not have any diversion airport
+        if (!$diversion_airport) {
+            return;
+        }
+
+        $pirep->loadMissing('aircraft', 'flight', 'user');
+        $aircraft = $pirep->aircraft;
+        $flight = $pirep->flight;
+        $user = $pirep->user;
+
+        $has_vmsacars = Module::find('VMSAcars');
+
+        if ($has_vmsacars && $flight) {
```

Comment:
This is interesting. Maybe it's better to have an event that's sent out and then this code I can move into the ACARS module? Keeps it separated a bit

Reply:
yes that can also be an idea. It means that the acars module interacts with the flights creating new ones and everything, which is perhaps not its initial vocation but I can do that without problem if you want me to

Reply:
Thinking about it, maybe it doesn't matter if they have ACARS or not. I think another setting for `pireps.diversion_create_reposition_flight` would work to check instead. It doesn't really matter if free flights are enabled or not, it's just a new flight that's been created so it wouldn't be a "free" one. 

What do you think @FatihKoz ?

Reply:
We could definitely do that.
The orginal idea was to create them if there was no free flight to ensure that the pilot does not get stuck at an airport without a flight and without being able to move around from one airport to another but even if there are freeflights we can create it.

Reply:
Yeah, it's a really good idea

Reply:
Maybe we can use the setting but with something to override it.
Like if setting is enabled then we add the flight
If setting is disabled and free flight are also disabled and there is no flight from diversion airport then we add it anyway

Reply:
I think let's just go with the setting for now, and add an event. Then I can pull this code into the ACARS module